### ISSUE TITLE
Installation Failure: Updated Manjaro (Arch-based) Linux

### ISSUE THREAD

**Issue by whelanh** [eddelbuettel/rquantlib#175]
install.packages('RQuantLib') fails on R 4.3.0 on updated Manjaro (arch-based) Linux system (I suspect this is a recurring problem with compiler mismatch?):

```
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ -std=gnu++17 accepts -g... yes
checking for g++ -std=gnu++17 option to enable C++11 features... none needed
checking how to run the C++ preprocessor... g++ -std=gnu++17 -E
checking whether the compiler supports GNU C++... (cached) yes
checking whether g++ -std=gnu++17 accepts -g... (cached) yes
checking for g++ -std=gnu++17 option to enable C++11 features... (cached) none needed
checking for R... yes
checking for quantlib-config... yes
checking for Boost development files... yes
checking for minimal Boost version... yes
configure: creating ./config.status
config.status: creating src/Makevars
Configured RQuantlib 0.4.17
** libs
using C++ compiler: ‘g++ (GCC) 12.2.1 20230201’
using C++14
g++ -std=gnu++14 -I"/usr/include/R/" -DNDEBUG  -I'/home/hugh/R/x86_64-pc-linux-gnu-library/4.3/Rcpp/include' -I/usr/local/include   -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects  -I/usr/include -fopenmp -pthread  -I../inst/include -I. -DBOOST_NO_AUTO_PTR -fopenmp -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects  -c RcppExports.cpp -o RcppExports.o
In file included from /usr/include/ql/quantlib.hpp:10,
                 from ../inst/include/rquantlib_internal.h:23,
                 from ../inst/include/RQuantLib.h:22,
                 from RcppExports.cpp:4:
/usr/include/ql/any.hpp:40:20: error: ‘any’ has not been declared in ‘std’
   40 |         using std::any;                   // NOLINT(misc-unused-using-decls)
      |                    ^~~
/usr/include/ql/any.hpp:41:20: error: ‘any_cast’ has not been declared in ‘std’
```

**Comment by eddelbuettel** [comment]
Can you provide a bit more detail, please?  

I see g++ 12.2.1, that should be good.  But what version of QuantLib is this trying to build against?   

FWIW QuantLib 1.30 came out recently, I updated it for Debian (I am the maintainer for it there) and also in my PPA.  I made two tiny changes in the github repo of RQuantLib for.

The CRAN version of RQuantLib has not been updated for that, so it would likely be happiest with QuantLib 1.29.  Is that what you use, or do you have something older?

**Comment by whelanh** [comment]
Thank you so much for responding.  The version of QuantLib I have installed/compiled is 1.30-1 from AUR.  So perhaps that is the problem. Sounds like I should either downgrade to 1.29 and/or use the verision in your PPA.  I will give that a try.  Thank you for all of your work on this package. Kindest regards, Hugh

**Comment by eddelbuettel** [comment]
No, 1.30-1 is good!  While it does not explain the error you showed (which I never saw on similar compiler versions) you can just try `remotes::install_github("eddelbuettel/rquantlib")` though I fear it may also fail.  The next thing we may have to control for is Boost versions but you are likely also current.  In a nutshell, I do not know where 

```
/usr/include/ql/any.hpp:40:20: error: ‘any’ has not been declared in ‘std’
   40 |         using std::any;                   // NOLINT(misc-unused-using-decls)
      |                    ^~~
/usr/include/ql/any.hpp:41:20: error: ‘any_cast’ has not been declared in ‘std’
```

came from.  "It all works over here."

**Comment by whelanh** [comment]
You might be right about Boost versions.  I see I have boost 1.81.0-3.  I see from the boost website that there is a 1.82.0 version of boost; is that the one you used?

**Comment by eddelbuettel** [comment]
No I am also on 1.81.  Weird.

**Comment by whelanh** [comment]
Well I am happy to report success now.  It was not a Boost issue (I did not change my configuration).  I changed my installation of "quantlib" from AUR 1.30-1 to the other option on AUR of 'quantlib-git" version 1.30r12ga342b44ff-1. These AUR issues are probably why you are happy to be using Debian :-)  After doing that I was able to install RQuantLib normally using install.packages('RQuantLib')...albeit with some warnings 
( "RQuantLib 0.4.17 built with QuantLib version 1.31-dev. See https://www.quantlib.org for more on QuantLib.
Warning message: In compareVersion(qlver, "1.7") : NAs introduced by coercion") 

 Thank you for your help and patience. Kindest regards, Hugh
